migrate/20230919_add_runner_pool.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table(:github_runner_vm_pool) do
+      column :id, :uuid, primary_key: true, default: nil
+      column :size, :integer, null: false
+      column :label, :text, null: false
+    end
+
+    # Will be used for join table to get vms from pool
+    # Cleaner approach is adding a pool_id column to vm table.
+    # However, we want to introduce github runner related code
+    # as isolated as possible.
```

Comment:
I think it is very likely that we would use vm pools for things other than github (maybe not immediately, but after we have more extra capacity to spare, we can use pools for regular vm provisionings as well.

For that reason, I think it is OK to add a pool_id column to the vm table. It would simplify the design and it would make pools more general for other services to use. We can make the above table just "vm_pool" and decouple pools from github runners.

Reply:
I wouldn't recommend this if it would cause more complex approach to generalize the pools, but it seems like it would end up being in similar complexity.

Reply:
If our aim is to design more general pools, we must keep parameters such as vm_size, boot_image, location, and any other differentiators, rather than just the runner label.

Reply:
Yes, for the beginning those should be enough. we don't need to keep disk size information. We can replace label with vm_size.

Initially we would only populate the pool for `boot_image: "github-ubuntu-2204"` and `location: "github-runners"`. As we get more capacity, we can populate more pools.

Reply:
I think I agree with this. The initial design was using pool_id and the relation model was also easier to grasp. I will revert back to that model.